.github/workflows/reusable-acceptance.yml

```diff
@@ -0,0 +1,128 @@
+          - {runner: "1", test-packages: "controller example ingress-gateway"}
+          - {runner: "2", test-packages: "mesh-gateway metrics"}
+          - {runner: "3", test-packages: "partitions sync terminating-gateway"}
+          - {runner: "4", test-packages: "vault"}
+
+      fail-fast: true
+    steps:      
+      - name: Checkout code
+        uses: actions/checkout@v2
+      
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ inputs.go-version }}
+
+      - name: Setup go mod cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Install gotestsum
+        run: |
+          wget https://github.com/gotestyourself/gotestsum/releases/download/v"${{ inputs.gotestsum-version }}"/gotestsum_"${{ inputs.gotestsum-version }}"_linux_amd64.tar.gz
+          sudo tar -C /usr/local/bin -xzf gotestsum_"${{ inputs.gotestsum-version }}"_linux_amd64.tar.gz
+          rm gotestsum_"${{ inputs.gotestsum-version }}"_linux_amd64.tar.gz
+
+      - run: mkdir -p ${{ env.TEST_RESULTS }}
+
+      - name: go mod download
+        working-directory: ${{ inputs.directory }}
+        run: go mod download
+      
+      - name: Create kind clusters
+        run: |
+          kind create cluster --name dc1 --image kindest/node:${{ inputs.kind-version }}
+          kind create cluster --name dc2 --image kindest/node:${{ inputs.kind-version }}
+
+      # We have to run the tests for each package separately so that we can
+      # exit early if any test fails (-failfast only works within a single
+      # package).
+      - name: Run acceptance tests ${{ matrix.runner }}
+        working-directory: ${{ inputs.directory }}
+        if: github.repository_owner == 'hashicorp' # This prevents running on forks
+        run: |
+          exit_code=0
+          echo "Running packages: ${{ matrix.test-packages }}"
+          for pkg in $(echo ${{ matrix.test-packages }})
+          do
+            fullpkg="github.com/hashicorp/consul-k8s/${{ inputs.directory }}/${pkg}"
+            echo "Testing package: ${fullpkg}"
+            if ! gotestsum --jsonfile=jsonfile-${pkg////-} -- ${fullpkg} -p 1 -timeout 2h -failfast \
+              ${{ inputs.additional-flags }} \
+              -enable-enterprise \
+              -enable-multi-cluster \
+              -debug-directory=${{ env.TEST_RESULTS }}/debug \
+              -consul-k8s-image=${{ inputs.consul-k8s-image }} 
+            then
+              echo "Tests in ${pkg} failed, aborting early"
+              exit_code=1
+              break
+            fi
+          done
+          gotestsum --raw-command --junitfile "${{ env.TEST_RESULTS }}/gotestsum-report.xml" -- cat jsonfile*
+          exit $exit_code
+    
+      - name: Upload tests
```

Comment:
Github Actions does not have a pretty test page that you can look at so we will have to look at the artifacts for tests.

Debug output is added as an artifact if the tests fail.